<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <!-- Meta tags for social media banners, these should be filled in appropriatly as they are your "business card" -->
    <!-- Replace the content tag with appropriate information -->
    <meta name="description" content="DESCRIPTION META TAG" />
    <meta property="og:title" content="SOCIAL MEDIA TITLE TAG" />
    <meta
      property="og:description"
      content="SOCIAL MEDIA DESCRIPTION TAG TAG"
    />
    <meta property="og:url" content="URL OF THE WEBSITE" />
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X630-->
    <meta property="og:image" content="static/image/your_banner_image.png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:image:height" content="630" />

    <meta name="twitter:title" content="TWITTER BANNER TITLE META TAG" />
    <meta
      name="twitter:description"
      content="TWITTER BANNER DESCRIPTION META TAG"
    />
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X600-->
    <meta
      name="twitter:image"
      content="static/images/your_twitter_banner_image.png"
    />
    <meta name="twitter:card" content="summary_large_image" />
    <!-- Keywords for your paper to be indexed by-->
    <meta name="keywords" content="KEYWORDS SHOULD BE PLACED HERE" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>
      CellTreeQM: Reconstructing Cell Lineage Trees from Phenotypic Features via
      Metric Learning
    </title>
    <link rel="icon" type="image/x-icon" href="static/images/favicon.ico" />
    <link
      href="https://fonts.googleapis.com/css?family=Google+Sans|Noto+Sans|Castoro"
      rel="stylesheet"
    />

    <link rel="stylesheet" href="static/css/bulma.min.css" />
    <link rel="stylesheet" href="static/css/bulma-carousel.min.css" />
    <link rel="stylesheet" href="static/css/bulma-slider.min.css" />
    <link rel="stylesheet" href="static/css/fontawesome.all.min.css" />
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1/css/academicons.min.css"
    />
    <link rel="stylesheet" href="static/css/index.css" />

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="https://documentcloud.adobe.com/view-sdk/main.js"></script>
    <script defer src="static/js/fontawesome.all.min.js"></script>
    <script src="static/js/bulma-carousel.min.js"></script>
    <script src="static/js/bulma-slider.min.js"></script>
    <script src="static/js/index.js"></script>
    <!-- ── MathJax v3: render LaTeX in \( … \) / $$ … $$ ───────────────────── -->
    <script>
      window.MathJax = {
        tex: {
          inlineMath: [
            ["\\(", "\\)"],
            ["$", "$"],
          ], // 支持 \( … \) 与 $ … $
          displayMath: [
            ["\\[", "\\]"],
            ["$$", "$$"],
          ],
        },
        svg: { fontCache: "global" }, // SVG 输出，兼容性最好
      };
    </script>
    <script
      defer
      src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"
    ></script>
    <!-- ─────────────────────────────────────────────────────────────────────── -->
  </head>
  <body>
    <section class="hero">
      <div class="hero-body">
        <div class="container is-max-desktop">
          <div class="columns is-centered">
            <div class="column has-text-centered">
              <h1 class="title is-1 publication-title">
                CellTreeQM: Reconstructing Cell Lineage Trees from Phenotypic
                Features via Metric Learning
              </h1>
              <div class="is-size-5 publication-authors">
                <!-- Paper authors -->
                <span class="author-block">
                  <a href="https://www.kuangda.tech/" target="_blank"
                    >Da Kuang</a
                  >,</span
                >
                <span class="author-block">
                  <a href="https://nayutaqiu.github.io/" target="_blank"
                    >Guanwen Qiu</a
                  >,</span
                >
                <span class="author-block">
                  <a
                    href="https://www.bio.upenn.edu/people/junhyong-kim"
                    target="_blank"
                    >Junhyong Kim</a
                  >
                </span>
              </div>

              <div class="is-size-5 publication-authors">
                <span class="author-block"
                  >University of Pennsylvania<br />ICML 2025</span
                >
              </div>

              <div class="column has-text-centered">
                <div class="publication-links">
                  <!-- Arxiv PDF link -->
                  <span class="link-block">
                    <a
                      href="https://arxiv.org/pdf/<ARXIV PAPER ID>.pdf"
                      target="_blank"
                      class="external-link button is-normal is-rounded is-dark"
                    >
                      <span class="icon">
                        <i class="fas fa-file-pdf"></i>
                      </span>
                      <span>Paper</span>
                    </a>
                  </span>

                  <!-- Github link -->
                  <span class="link-block">
                    <a
                      href="https://github.com/YOUR REPO HERE"
                      target="_blank"
                      class="external-link button is-normal is-rounded is-dark"
                    >
                      <span class="icon">
                        <i class="fab fa-github"></i>
                      </span>
                      <span>Code</span>
                    </a>
                  </span>
                  <!-- Dataset -->
                  <span class="link-block">
                    <a
                      href=""
                      target="_blank"
                      class="external-link button is-normal is-rounded is-dark"
                    >
                      <span
                        class="icon"
                        style="vertical-align: middle; font-size: 20px"
                        >&#129303;</span
                      >
                      <span style="vertical-align: middle">Dataset</span>
                    </a>
                  </span>
                  <!-- ArXiv abstract Link -->
                  <span class="link-block">
                    <a
                      href="https://arxiv.org/abs/<ARXIV PAPER ID>"
                      target="_blank"
                      class="external-link button is-normal is-rounded is-dark"
                    >
                      <span class="icon">
                        <i class="ai ai-arxiv"></i>
                      </span>
                      <span>arXiv</span>
                    </a>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>

    <!-- Teaser video-->
    <section class="hero teaser">
      <div class="container is-max-desktop">
        <div class="hero-body has-text-centered">
          <img
            src="static/images/teaser_image_1.png"
            alt="Teaser Image"
            class="teaser-image_1"
          />
          <h2 class="subtitle has-text-centered">
            Exploring phenotype-based cell lineage reconstruction. This figure
            highlights the focus of our study on reconstructing cell lineage
            trees using phenotype data, specifically gene expression profiles
            (right), in contrast to traditional methods that rely on genotype
            data, such as DNA sequences (left).
          </h2>
        </div>
      </div>
    </section>
    <!-- End teaser video -->

    <!-- Paper abstract -->
    <section class="section hero is-light">
      <div class="container is-max-desktop">
        <div class="columns is-centered has-text-centered">
          <div class="column is-four-fifths">
            <h2 class="title is-3">Abstract</h2>
            <div class="content has-text-justified">
              <p>
                How a single fertilized cell gives rise to a complex array of
                specialized cell types in development is a central question in
                biology. The cells replicate to generate cell lineages and
                acquire differentiated characteristics through poorly understood
                molecular processes. A key approach to studying developmental
                processes is to infer the tree graph of cell lineage histories,
                which provides an analytical framework for dissecting individual
                cells' molecular decisions during replication and
                differentiation (i.e., acquisition of specialized traits).
                Although genetically engineered lineage-tracing methods have
                advanced the field, they are either infeasible or ethically
                constrained in many organisms. By contrast, modern single-cell
                technologies can measure high-content molecular profiles (e.g.,
                transcriptomes) in a wide range of biological systems. Here, we
                introduce CellTreeQM, a novel deep learning method based on
                transformer architectures that learns an embedding space with
                geometric properties optimized for tree-graph inference. By
                formulating the lineage reconstruction problem as tree-metric
                learning, we systematically explore weakly supervised training
                settings at different levels of information and present the Cell
                Lineage Reconstruction Benchmark to facilitate comprehensive
                evaluation. This benchmark includes (1) synthetic data modeled
                via Brownian motion with independent noise and spurious signals;
                (2) lineage-resolved single-cell RNA sequencing datasets.
                Experimental results show that CellTreeQM recovers lineage
                structures with minimal supervision and limited data, offering a
                scalable framework for uncovering cell lineage relationships. To
                our knowledge, this is the first method to cast cell lineage
                inference explicitly as a metric learning task, paving the way
                for future computational models aimed at uncovering the
                molecular dynamics of cell lineage.
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <!-- End paper abstract -->

    <section class="section hero is-small">
      <div class="container is-max-desktop">
        <div class="columns is-centered">
          <div class="column is-full">
            <div class="content">
              <h2 class="title is-3">Overview of the CellTreeQM workflow</h2>
              <p>
                When the full tree is known as prior knowledge, this is a
                supervised setting. When no prior information about the tree is
                available, the setting is unsupervised. In between, we highlight
                two weakly supervised settings: the High-level Partitioning
                Setting, where only high-level groupings are available, and the
                Partially Leaf-labeled Setting, where topological labels are
                provided for a subset of leaves.
              </p>
              <p>
                <img
                  src="static/images/teaser_image_2.png"
                  alt="Directional Weight Score"
                  class="blend-img-background center-image"
                  style="max-width: 100%; height: auto"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>

    <section class="section hero is-small is-light">
      <div class="container is-max-desktop">
        <div class="columns is-centered">
          <div class="column is-full">
            <div class="content">
              <h2 class="title is-3">Four-Point Condition and Quartet Loss</h2>
              <div class="level-set has-text-justified">
                <p>
                  <strong>Buneman’s theorem&nbsp;(1971)</strong> states that for
                  any four leaves <em>A, B, C, D</em>, the three distance sums
                  \(S_1=d_{AB}+d_{CD}\), \(S_2=d_{AC}+d_{BD}\),
                  \(S_3=d_{AD}+d_{BC}\) satisfy:
                  <em
                    >exactly two of them match and both exceed the remaining
                    one</em
                  >
                  <span class="has-text-grey">(right diagram)</span>. This law
                  both detects additivity and uniquely determines the unrooted
                  quartet topology.
                </p>
                <p>
                  <img
                    src="static/images/4point_condition.png"
                    alt="Sizes of model trees"
                    class="blend-img-background center-image"
                    style="max-width: 100%; height: auto"
                  />
                </p>
                <p>
                  In <em>CellTreeQM</em>, we turn the theorem into a loss:
                  \(L_\text{close}=|S_1-S_2|\) forces the two largest sums to
                  coincide, while
                  \(L_\text{push}=\bigl[S_3-\tfrac{S_1+S_2}{2}+m_0\bigr]_+\)
                  keeps the smallest sum sufficiently lower. Minimising
                  \(L_\text{additivity}=L_\text{close}+L_\text{push}\)
                  encourages the latent space to respect tree geometry even when
                  the full lineage is unknown.
                </p>
                <div class="hero-body has-text-centered">
                  <img
                    src="static/images/4point_condition_1.png"
                    alt="Sizes of model trees"
                    class="blend-img-background center-image"
                    style="max-width: 30%; height: auto"
                  />
                </div>
                <p>
                  This figure illustrates the geometric intuition of the loss.
                  When additivity is violated, the structure of a quartet can be
                  imagined as a “box” with an extra edge. The \(L_\text{close}\)
                  term encourages the top two distance sums to become more
                  similar, thereby reducing the imbalance that creates the
                  box-like distortion. In effect, it ensures that the box is not
                  “fat”. Meanwhile, the \(L_\text{push}\) term increases the gap
                  between the smallest sum and the average of the top two,
                  effectively “widening the bridge.” This widening enhances the
                  tree model’s robustness to noise and distortions from the
                  ideal tree structure.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>

    <section class="section hero is-small">
      <div class="container is-max-desktop">
        <div class="columns is-centered">
          <div class="column is-full">
            <div class="content">
              <h2 class="title is-3">High-Level Partitioning Results</h2>
              <div class="level-set has-text-justified">
                <p>
                  In the high-level partitioning setting, we assume that only
                  coarse clade groupings are known (e.g., the first few
                  root-level splits), without full tree labels. CellTreeQM uses
                  this partial supervision to infer quartet relationships. Our
                  model significantly outperforms contrastive baselines
                  (Triplet, Quadruplet) across all levels of priors. Notably,
                  CellTreeQM recovers known tree structures with near-perfect
                  accuracy and generalizes well to unseen quartets.
                </p>
                <p>
                  <img
                    src="static/images/results_1.png"
                    alt="Directional Weight Score"
                    class="blend-img-background center-image"
                    style="max-width: 100%; height: auto"
                  />
                </p>
                <p></p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <!--BibTex citation -->
    <section class="section" id="BibTeX">
      <div class="container is-max-desktop content">
        <h2 class="title">BibTeX</h2>
        <pre><code>@article{kuang2025reconstructingcelllineagetrees,
        title={Reconstructing Cell Lineage Trees from Phenotypic Features with Metric Learning}, 
        author={Da Kuang and Guanwen Qiu and Junhyong Kim},
        year={2025},
        eprint={2503.13925},
        archivePrefix={arXiv},
        primaryClass={cs.LG},
        url={https://arxiv.org/abs/2503.13925}, 
  }</code></pre>
      </div>
    </section>
    <!--End BibTex citation -->

    <footer class="footer">
      <div class="container">
        <div class="columns is-centered">
          <div class="column is-8">
            <div class="content">
              <p>
                This page was built using the
                <a
                  href="https://github.com/eliahuhorwitz/Academic-project-page-template"
                  target="_blank"
                  >Academic Project Page Template</a
                >
                which was adopted from the <a
                  href="https://nerfies.github.io"
                  target="_blank"
                  >Nerfies</a
                > project page. You are free to borrow the source code of this
                website, we just ask that you link back to this page in the
                footer. <br />
                This website is licensed under a
                <a
                  rel="license"
                  href="http://creativecommons.org/licenses/by-sa/4.0/"
                  target="_blank"
                  >Creative Commons Attribution-ShareAlike 4.0 International
                  License</a
                >.
              </p>
            </div>
          </div>
        </div>
      </div>
    </footer>

    <!-- Statcounter tracking code -->

    <!-- You can add a tracker to track page visits by creating an account at statcounter.com -->

    <!-- End of Statcounter Code -->
  </body>
</html>
